components/feature/session/src/main/java/mozilla/components/feature/session/SessionFeature.kt

```diff
@@ -32,9 +32,13 @@ class SessionFeature(
      * @return true if the event was handled, otherwise false.
      */
     fun handleBackPressed(): Boolean {
-        sessionManager.selectedSession?.let { session ->
-            if (session.canGoBack) {
-                sessionUseCases.goBack.invoke()
+        val session = sessionId?.let {
+            sessionManager.findSessionById(it)
+        } ?: sessionManager.selectedSession
+
+        session?.let { it ->
+            if (it.canGoBack) {
+                sessionUseCases.goBack.invoke(it)
                 return true
             }
```

Comment:
nit: Since the `session` reference is immutable you do not really need `let` here (since Kotlin can smartcast), e.g.:

```Kotlin
if (session != null && session.canGoBack) {
    sessionUseCases.goBack.invoke(session)
}
```

or maybe:

```kotlin
if (session?.canGoBack == true) {
    sessionUseCases.goBack.invoke(session)
}
```

Reply:
Nice, I like the latter (because no nulls). :)

Reply:
I agree that `null` can be a pain. But your refusal to even type it is hilarious :)